yt_dlp/extractor/kukululive.py

```diff
@@ -0,0 +1,125 @@
+from .common import InfoExtractor
+from ..utils import (
+    ExtractorError,
+    clean_html,
+    get_element_by_id,
+    urljoin,
+)
+from ..compat import compat_parse_qs
+
+
+class KukuluLiveIE(InfoExtractor):
+    _VALID_URL = r'https?://live\.erinn\.biz/live\.php\?h(?P<id>\d+)'
+    _TESTS = [{
+        'url': 'https://live.erinn.biz/live.php?h675134569',
+        'only_matching': True,
+    }]
+
+    def _get_quality_meta(self, video_id, desc, code, force_h264=''):
+        description = desc if force_h264 == '' else f'{desc} (force_h264)'
+        qs = self._download_webpage(
+            'https://live.erinn.biz/live.player.fplayer.php', video_id,
+            query={
+                'hash': video_id,
+                'action': f'get{code}liveByAjax',
+                'force_h264': force_h264,
+            },
+            note=f'Downloading {description} quality metadata',
+            errnote=f'Unable to download {description} quality metadata')
+        return compat_parse_qs(qs)
+
+    def _add_quality_formats(self, formats, quality_meta):
+        vcodec = quality_meta.get('vcodec')[0]
+        quality = quality_meta.get('now_quality')[0]
+        quality_priority = {
+            'high': 3,
+            'h264': 2,
+            'low': 1,
+        }
+        formats.extend([
+            {
+                'format_id': quality,
+                'url': quality_meta.get('hlsaddr')[0],
+                'ext': 'mp4',
+                'vcodec': vcodec,
+                'quality': quality_priority[quality],
+            },
+            {
+                'format_id': f'{quality}-rtmp',
+                'url': quality_meta.get('streamaddr')[0],
+                'ext': 'mp4',
+                'vcodec': vcodec,
+                'quality': quality_priority[quality] - 5,
+            },
+            {
+                'format_id': f'{quality}-audioonly',
+                'url': quality_meta.get('hlsaddr_audioonly')[0],
+                'ext': 'aac',
+                'vcodec': 'none',
+                'quality': quality_priority[quality] - 10,
+            },
+        ])
```

Comment:
From my testing, RTMP doesn't work. Can you report something different?
```suggestion
        vcodec = traverse_obj(quality_meta, ('vcodec', 0))
        quality = traverse_obj(quality_meta, ('now_quality', 0))
        quality_priority = {
            'high': 3,
            'h264': 2,
            'low': 1,
        }.get(quality, 0)
        if traverse_obj(quality_meta, ('hlsaddr', 0, {url_or_none})):
            formats.append({
                'format_id': quality,
                'url': quality_meta['hlsaddr'][0],
                'ext': 'mp4',
                'vcodec': vcodec,
                'quality': quality_priority,
            })
        if traverse_obj(quality_meta, ('hlsaddr_audioonly', 0, {url_or_none})):
            formats.append({
                'format_id': f'{quality}-audioonly',
                'url': quality_meta['hlsaddr_audioonly'][0],
                'ext': 'm4a',
                'vcodec': 'none',
                'quality': quality_priority,
            })
```

Reply:
Didn't actually test RTMP